src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java

```diff
@@ -576,10 +590,76 @@ public void onUpgradeComplete(final int newLevel)
           workOrder.isMirrored());
         this.setHeight(wrapper.getHeight());
         this.setCorners(corners.getFirst().getFirst(), corners.getFirst().getSecond(), corners.getSecond().getFirst(), corners.getSecond().getSecond());
+
+        // firework spawning possibly maybe
+
+        //ItemStack items = new ItemStack(f);
+
+
+
+        fireWorkCounter++;
+
+        if (fireWorkCounter % 2 == 1) {
```

Comment:
can you use `newLevel` variable instead?
Also, remove unneeded comments and loggers